pandas/core/arrays/array_.py

```diff
@@ -0,0 +1,129 @@
+import numpy as np
+
+from pandas._libs import lib, tslibs
+
+from pandas.core.dtypes.common import is_extension_array_dtype
+from pandas.core.dtypes.dtypes import registry
+from pandas.core.dtypes.generic import ABCIndexClass, ABCSeries
+
+from pandas import compat
+
+
+def array(data, dtype=None, copy=False):
+    """
+    Create an array.
+
+    .. versionadded:: 0.24.0
+
+    Parameters
+    ----------
+    data : Sequence[object]
+        A sequence of objects. The scalars inside `data` should
+        be instances of the scalar type for `dtype`.
+
+        When `data` is an Index or Series, the underlying array
+        will be extracted from `data`.
+
+    dtype : Union[str, np.dtype, ExtensionDtype], optional
+        The dtype to use for the array. This may be a NumPy
+        dtype or an extension type registered with pandas using
+        :meth:`pandas.api.extensions.register_extension_dtype`.
+
+        If not specified, there are two possibilities:
+
+        1. When `data` is a :class:`Series`, :class:`Index`, or
+           :class:`ExtensionArray`, the `dtype` will be taken
+           from the data.
+        2. Otherwise, pandas will attempt to infer the `dtype`
+           from the data.
+
+        Note that when `data` is a NumPy array, ``data.dtype`` is
```

Comment:
this is not true, it is passed onto np.array

Reply:
This is trying to convey the fact that

```python
np.array([pd.Period('2000', 'D')], dtype=object)
```

will return a PeriodArray, rather than an object-dtype ndarray. I'll try to clarify that an ndarray's `.dtype` is ignored *for inference*.

Reply:
right, ok